### ISSUE TITLE
404 when installing private scoped package from npm registry

### ISSUE THREAD

**Issue by kmoe** [npm/npm#18569]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [x] npm is producing an incorrect install.
  - [x] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

I have published a scoped package, `@X/Y`, to the main npm registry. This is a private package, restricted to members of the npm organisation `@X`.

When I run `npm install` on package `Z` which has `@X/Y` as a dependency:
 - it works on my own machine, which is the one I used to publish the package (presumably because the tarball is in the local cache)
 - it doesn't work on my colleague's machine or the CI machine, even after `npm login` and verifying the `.npmrc` is identical.

The error I get is:
```
2840 http fetch GET 404 https://registry.npmjs.org/@tallarium%2fott 126ms
2841 silly fetchPackageMetaData error for @tallarium/ott@^0.0.2 404 Not Found: @tallarium/ott@^0.0.2
2842 verbose stack Error: 404 Not Found: @tallarium/ott@^0.0.2
```

Please see full npm-debug.log [here](https://gist.github.com/kmoe/97dadffd48692bf393a7aa6af0245635).

#### How can the CLI team reproduce the problem?

0. Publish a private scoped package to the npm registry.
1. From a different machine (or having cleared the npm cache), attempt to install the private scoped package.

### supporting information:

 - `npm -v` prints: 5.3.0
 - `node -v` prints: v8.4.0
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Linux
 - Network issues: N/A
 - Container: N/A

#### note

There are several issues regarding this already, but it looks like they are being closed by the bot before the potential bug is triaged.

For example:

 - https://github.com/npm/npm/issues/8770 (closed by bot, incorrectly labelled as support)
 - https://github.com/npm/npm/issues/16180 (closed by bot after no response from maintainers)

**Comment by dancrumb** [comment]
I'm also seeing this while trying to build a docker image. On each run, it's a different package that I get a 404 for, but they're always scoped packages.

**Comment by genericlady** [comment]
This isn't working for me even on the machine i used to publish the privately scoped module.

I followed this tut https://docs.npmjs.com/private-modules/intro

```
♥ npm access restricted @stuffandthings/redux-localstorage-plus                                    
npm ERR! Windows_NT 10.0.10586                                                               
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\
pm\\bin\\npm-cli.js" "access" "restricted" "@stuffandthings/redux-localstorage-plus"               
npm ERR! node v6.11.3                                                                        
npm ERR! npm  v3.10.10                                                                       
npm ERR! code E404                                                                           
                                                                                             
npm ERR! 404 Registry returned 404 for POST on https://registry.npmjs.org/-/package/%40stuffandthings
%2Fredux-localstorage-plus/access                                                            
                                                                                             
npm ERR! Please include the following file with any support request:                         
npm ERR!     C:\Users\stuffandthings\Code\react\redux-localstorage-plus\npm-debug.log              
redux-localstorage-plus                                                                      
♥                                                                                            
```

**Comment by hlummis** [comment]
I'm having the same issue. It looks like when I remove the .npmrc file from my application directory, updates/installs work as expected. Considering I need the .npmrc to build my app in docker, this won't solve my problem, but it might help in diagnosing what's wrong...

**Comment by tfoel** [comment]
Surprising that this one has received no attention from the project devs.
NPM basically fails if you try to mix in private scoped packages, making it hard to use in company environment.

**Comment by ArcanisCz** [comment]
I have similar error, but exactly when running against public npm. Running trhough our nexus solves the issue.

When running with verbose, it prints `npm http fetch GET 404 https://registry.npmjs.org/@storybook%2faddon-info 830ms
`. There, i suppose, that encoded / character makes the trouble, because if i try exact url in browser, it works. And i suppose, our Nexus will fetch correct url.

(Windows 10, tried UTF and CP1250 of package json. Npm 5.6.0)

**Comment by samslater** [comment]
Having this same issue.  Cannot remove the .npmrc file as I need to install packages from private registry as well as public.  Tried adding `@babel:registry=https://registry.npmjs.org/` to `.npmrc` file but still getting 404 when attempting to install @babel scoped packages.  Anybody have a solution?

**Comment by ArcanisCz** [comment]
Yesterday, i came across new info! As this happened to my new computer, not associated by any development in the past, freshly installed npm/node, and could not install scoped package. It was weird to me, so i dug again on the internet and was able to install that packages by doing **npm login** first!

Its very strange why you have to be signed up for acces to public packages and it even give you some cryptic erorr...

**Comment by daniel-livestories** [comment]
@ArcanisCz, you are absolutely correct.  This fixed my issue.  Unfortunate that npm returns a 404 for an auth issue.

**Comment by BobbyJohansen** [comment]
any movement on this? its really annoying not to be able to just login to a private repo and publish

**Comment by bpluly** [comment]
Reading through this it looks like there are at least two slightly different issues.
Both need reproducing with clear examples.

1. For some circumstances the login state is either lost or the user hasn't logged in so that access permission is granted for the private package.
2. Special characters for some circumstances are escaped, / and @ are two that are instanced.

Could someone that has the correct rights add a triage_needed label?

**Comment by GeoffreyBooth** [comment]
I was having this issue too, but logging in via `npm login` fixed it for me. I was trying to download only packages from the public NPM registry.

**Comment by yizziy** [comment]
Having the same problem with scoped repositories not being found, I was checking the authToken in my .npmrc and compared it to the ones I have on npmjs.com . Seems like my Token was no longer available. After I created a new one and added this to the npmrc the package was installed correctly.  
Maybe there is something going on with tokens getting invalidated ?
